nnstreamer/tensor_filter/tensor_filter_nntrainer.cc

```diff
@@ -239,20 +206,39 @@ static int nntrainer_run(const GstTensorFilterProperties *prop,
   return nntrainer->run(input, output);
 }
 
-static int nntrainer_getInputDim(const GstTensorFilterProperties *prop,
-                                 void **private_data, GstTensorsInfo *info) {
+static int nntrainer_setInputDim(const GstTensorFilterProperties *prop,
+                                 void **private_data,
+                                 const GstTensorsInfo *in_info,
+                                 GstTensorsInfo *out_info) {
   NNTrainerInference *nntrainer =
     static_cast<NNTrainerInference *>(*private_data);
-  g_return_val_if_fail(nntrainer && info, -EINVAL);
-  return nntrainer->getInputTensorDim(info);
-}
+  g_return_val_if_fail(prop && nntrainer && in_info && out_info, -EINVAL);
 
-static int nntrainer_getOutputDim(const GstTensorFilterProperties *prop,
-                                  void **private_data, GstTensorsInfo *info) {
-  NNTrainerInference *nntrainer =
-    static_cast<NNTrainerInference *>(*private_data);
-  g_return_val_if_fail(nntrainer && info, -EINVAL);
-  return nntrainer->getOutputTensorDim(info);
+  auto num_input = in_info->num_tensors;
+  g_return_val_if_fail(num_input != 0, -EINVAL);
+
+  nntrainer->setBatchSize(in_info->info[0].dimension[3]);
```

Comment:
this statement might be redundant.

Reply:
If you want to setBatchSize here, use setProperty() to do so. `setBatchSize()` will soon become private.

Reply:
> this statement might be redundant.

Could you elaborate on this?

> If you want to setBatchSize here, use setProperty() to do so. setBatchSize() will soon become private.

Done.

Reply:
> this statement might be redundant.

Because inference will check the `batchsize` of the data and reset the `batchsize` if it is different. Setting batchsize here does not provide any speedup as well, because memory is allocated on the first call of inference.

Reply:
Understood, it might be better if we could make it allocate at this stage.
Just to expose the intention, let me leave this as it is.

Reply:
Alright. 
FYI: you can call `NeuralNetwork::allocate(false)` after this step (this has the strong assumption that network properties are fixed). This will allocate at this stage outside, reducing the overhead from the first call of the `NeuralNetwork::inference()`. 
Note that this will reduce the possibility of using the ml::train::model as model::allocate() might not get exposed with c++ API.

Reply:
Document about it :+1: